src/logger.js

```diff
@@ -0,0 +1,45 @@
+import CoreObject from './metal/core-object';
+
+const Logger = CoreObject.extend({
+  /**
+   * Wrapper around the console log so in the future we can have better dev output.
+   * Also allows us to disable output in production.
+   * @class Logger
+   * @constructor
+   */
+  constructor() {
+  },
+
+  /* eslint-disable no-console */
+  log() {
+    console.log(...this.tagLog(...arguments));
```

Comment:
This may be one level too deep abstraction, but you could have a high order function that returns these:

```javascript

wrapConsole(logCommand) {
  return function() {
    console[logCommand](...this.tagLog(...arguments));
  };
}


// Then in your class
CoreObject.extend({
  log: wrapConsole('log'),
  debug: wrapConsole('debug')
})
```

Also, a thing I'm opinionated about (so it may be the wrong way to go): I don't like presenting the default console.log. I feel like if we're outputting something, we should be opinionated about what that something is. It's one of debug info, informative, warning, or errors. `console.log` is very ambiguous on intent.